core/adminserver/admin_server.go

```diff
@@ -348,3 +348,37 @@ func (s *Server) DeleteDomain(ctx context.Context, in *pb.DeleteDomainRequest) (
 func (s *Server) UndeleteDomain(ctx context.Context, in *pb.UndeleteDomainRequest) (*google_protobuf.Empty, error) {
 	return nil, status.Errorf(codes.Unimplemented, "not implemented")
 }
+
+// GarbageCollect looks for domains that have been deleted for longer than a given duration and fully deletes them.
+func (s *Server) GarbageCollect(ctx context.Context, in *pb.GarbageCollectRequest) (*pb.GarbageCollectResponse, error) {
+	before, err := ptypes.Timestamp(in.GetBefore())
+	if err != nil {
+		return nil, err
+	}
+
+	// Search for domains older than in.Duration.
```

Comment:
s/older than in.Duration/deleted earlier than in.Before/

Reply:
done